crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1987,9 +1991,25 @@ fn conv_sort_path(
         .iter()
         .map(|t| conv_sort(genv, t, next_infer_sort))
         .try_collect()?;
+
     Ok(rty::Sort::app(ctor, args))
 }
 
+fn path_generics_correct(path: &fhir::SortPath, sort_generics: rty::Generics) -> bool {
+    // Check that the number of generics for a path are correct
+    //
+    // Note: We include base when collecting generics (i.e. T as base) for a def id
+    // because often times there are refinements over these types (see extern spec for iter
+    // as an example)
+    //
+    if path.args.len() != sort_generics.own_default_count_with_base() {
```

Comment:
@vrindisbacher A sort doesn't necessarily have the same number of generics as the type it corresponds to. For example, if you write
```rust
#[flux::refined_by(keys: Set<K>)
struct Map<K, V> { ... }
```
we will generate a sort with a single generic. The mapping between type parameters and sort parameters is tracked in [AdtSortDef](https://flux-rs.github.io/flux/doc/flux_middle/rty/struct.AdtSortDef.html). Here you are basically recomputing that mapping but we could use the information in `AdtSortDef` directly.

Reply:
@nilehmann, so if I understand correctly, we want to make sure that we are not using more generics than are defined for the type it corresponds to?

Reply:
At this point, we can forget there's a corresponding type. We only want to look at the sort declaration.

I also just noticed that we need to do this for all sorts not just adt sorts. The syntax will for example let you write `int<bool>` which is nonsense.

* `Set` must have 1 argument
* `Map` must have 2
* For adt sorts we get the information from the `AdtSortDef`. The number of parameters is the length of https://flux-rs.github.io/flux/doc/flux_middle/rty/struct.AdtSortDefData.html#structfield.params, we must expose that value in a method.
* all other sorts must have zero arguments.

Reply:
And the error should say something like

```
error[E0999]: sort takes 0 arguments but 1 argument was supplied
 --> ../sort_gen.rs:8:1
   |
10 |     fn foo(s: S<int>) -> int {
   |               ^
   |               |
   |               expected 0 arguments
 ```
 
For the span, I think we can take the span of the ident in the last segment of the `SortPath`

Reply:
@nilehmann, went ahead and updated this.